mdal/frmts/mdal_gdal.cpp

```diff
@@ -349,7 +355,8 @@ void MDAL::DriverGdal::addDataToOutput( GDALRasterBandH raster_band, std::shared
     {
       unsigned int idx = x + mXSize * y;
       double val = mPafScanline[x];
-      if ( !MDAL::equals( val, nodata ) )
+
+      if ( std::isnan( nodata ) || !MDAL::equals( val, nodata ) )
```

Comment:
It took me a bit to understand what is going on... maybe it would be nicer to have bool "hasNoData" variable instead of check for isnan(nodata)

Reply:
+1